src/sql/azdata.proposed.d.ts

```diff
@@ -2011,6 +2011,26 @@ declare module 'azdata' {
 			 */
 			isPrimary: boolean;
 		}
+
+		export interface FileFilters {
+			/**
+			 * The label to display in the file filter field next to the list of filters.
+			 */
+			label: string;
+			/**
+			 * The filters to limit what files are visible in the file browser (e.g. '*.sql' for SQL files).
+			 */
+			filters: string[];
+		}
+
+		/**
+		 * Opens a file dialog to select a file path on the specified server's machine.
+		 * @param connectionUri The URI of the connection to the target server
+		 * @param targetPath The file path on the server machine to open by default in the dialog
+		 * @param fileFilters The filters used to limit which files are displayed in the file browser
+		 * @returns The path of the file chosen from the dialog, and undefined if the dialog is closed without selecting anything.
+		 */
+		export function openFileBrowserDialog(connectionUri: string, targetPath: string, fileFilters: FileFilters[]): Thenable<string | undefined>;
```

Comment:
Maybe `openServerFileBrowserDialog` to clearly indicate that it's for browsing server resources, not local ones? I'd also add in the description that showOpenDialog should be used for browsing local resources.